### PR TITLE
Enforce AUTH_REVOCABLE on contract balances

### PR DESCRIPTION
**Pull request by sisuresh**
### What

Enforce AUTH_REVOCABLE on contract balances to match classic and SAC account balance behavior.

### CODE REVIEW

File: soroban-env-host/src/test/token.rs [hunk 1]
```diff
@@ -991,6 +991,56 @@ fn test_clawback_on_contract() {
...
+
+    // set auth revocable
+    test.update_account_flags(&issuer_ledger_key, AccountFlags::RevocableFlag as u32);
+
+    // Now auth can be revoked
+    token
+        .set_authorized(&admin, user_1_addr.clone(), false)
+        .unwrap();
+
+    assert!(!token.authorized(user_1_addr.clone()).unwrap());
```
Comment (dmkozh) [inline soroban-env-host/src/test/token.rs:1032]: Do we have coverage for making sure that ops actually don't succeed?

**Review by dmkozh — APPROVED** [review]

File: soroban-env-host/src/test/token.rs (hunk 1 above)
Comment (sisuresh) [reply soroban-env-host/src/test/token.rs:1032]: I don't think I understand your question.

Comment (dmkozh) [reply soroban-env-host/src/test/token.rs:1032]: Do we have a test that does `token.transfer(user_1_addr,...)` when `user_1_addr` is a deauthorized contract ?

Comment (sisuresh) [reply soroban-env-host/src/test/token.rs:1032]: It looks like we're missing that for contract balances (I think we're missing quite a few test cases for contract balances actually).

Comment (dmkozh) [reply soroban-env-host/src/test/token.rs:1032]: Right, so can we add it?

Comment (sisuresh) [reply soroban-env-host/src/test/token.rs:1032]: Ah I see what you originally meant. I have a list of tests we need to add, and would prefer to do this when we address the missing tests (I need to open a github issue with that list.)

Comment (dmkozh) [reply soroban-env-host/src/test/token.rs:1032]: Ok, SG.

**Review by dmkozh — APPROVED** [review]

**Pull request merged**